### ISSUE TITLE
[Bug]: Hosting Configuration "please wait" spinner is stuck forever after reverting

### ISSUE THREAD

**Issue by p-jackson** [Automattic/wp-calypso#61974]
### Quick summary

The Hosting Configuration screen can be used to initiate AT. _If_ this is the method that's used to initiate AT _and_ the atomic site is reverted at a later date, then the Hosting Configuration screen gets stuck showing an incorrect message and a spinner that doesn't go away.

### Steps to reproduce

1. Start with a simple business site
2. `/hosting-config/{{ site slug }}`
3. Activate hosting features
4. After AT is complete, revert the site back to a simple site
5. Go back and look at `/hosting-config/{{ site slug }}`

### What you expected to happen

Should see the `Please activate the hosting access to begin using these features.` message, since we've reverted back to simple we'd need to AT again to access them.

<img width="1197" alt="Screenshot 2022-03-16 at 7 41 45 PM" src="https://user-images.githubusercontent.com/1500769/158531597-09f6775e-e928-4991-9684-136bc8fbdca2.png">

### What actually happened

The `Please wait while we activate the hosting features.` message spins forever.

https://user-images.githubusercontent.com/1500769/158531366-28d3e269-2d40-44f9-8201-4418b9c3c9ce.mov

The site does not actually appear to be transferring back to atomic again.

### Simple/Atomic

Atomic

### Other notes

I suspect there's a piece of state that's been stored in the Redux store that remembers we clicked the activate button but isn't being reset. When the Redux store is cleared by re-logging in (or by switching to another Redux store by testing on `calypso.localhost:3000`) the bug is fixed.

### Reproducibility

Consistent

### Severity

Some (< 50%)

### Available workarounds?

Yes, easy to implement

### Workaround details

Log out and log in again.
This clears the locale redux store, which is probably why Calypso thinks it's stuck in the state of activating the hosting features

**Comment by JoshuaGoode** [comment]
Was the site fully reverted at this time?

We place restrictions on a site to prevent it from going Atomic again until the entire revert and lossless import completes.

Right now, this can take some time.

**Comment by p-jackson** [comment]
Turns out we're not actually polling, I misunderstood the network traffic. We're actually showing the spinner but not checking with the server at all about whether "transfer is complete".

> Was the site fully reverted at this time?

I'm not sure whether it was fully reverted or not, is there somewhere I can see whether a particular site is still in the backlog?

Either way, Calypso isn't actually checking with the backend so the spinner state isn't reflecting the site's eligibility to start a transfer again. I think the better way to handle this would be to allow the user to click "Activate" again immediately, but for the AT eligibility checks to fail if it's still in the backlog. So the same place that checks whether they have an appropriate plan etc. wouldn't let them proceed if it's still in the backlog.

<img width="744" alt="Screenshot 2022-03-18 at 11 15 36 AM" src="https://user-images.githubusercontent.com/1500769/158904146-b29f9962-8e69-4e58-ba0d-e818bb7e847c.png">

That way we've got more room for a message than the single line element with the spinner.
